File: model/src/main/java/io/github/runtimemodels/chazm/event/AchievesEventFactory.java
package io.github.runtimemodels.chazm.event;

import io.github.runtimemodels.chazm.relation.Achieves;

/**
 * The {@linkplain AchievesEventFactory} interface defines the API for constructing {@linkplain AchievesEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface AchievesEventFactory {

    /**
     * Constructs an {@linkplain AchievesEvent}.
     *
     * @param category the {@linkplain EventCategory}.
     * @param achieves the {@linkplain Achieves}.
     * @return an {@linkplain AchievesEvent}.
     */
    AchievesEvent build(EventCategory category, Achieves achieves);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/AchievesEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.relation.Achieves

/**
 * The [AchievesEventFactory] interface defines the API for constructing [AchievesEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface AchievesEventFactory {

    /**
     * Constructs an [AchievesEvent].
     *
     * @param category the [EventType].
     * @param achieves the [Achieves].
     * @return an [AchievesEvent].
     */
    fun build(category: EventType, achieves: Achieves): AchievesEvent

}
